src/root/async.c

```diff
@@ -109,7 +114,8 @@ unsigned __stdcall startthread(void *p)
     AsyncRead *aw = (AsyncRead *)p;
 
     //printf("aw->filesdim = %p %d\n", aw, aw->filesdim);
-    for (size_t i = 0; i < aw->filesdim; i++)
+    size_t dim = aw->filesdim;
+    for (size_t i = 0; i < dim; i++)
```

Comment:
Ketmar wrote: "'cause `AsyncRead::dispose` can `free()` aw before loop in `startthread` completes, and then `startthread` will try to reference already freed memory in loop condition check. this is very unlikely, but still possible situation. so we have to cache the value."

Please add this as a comment.